artifact/v3/scripts/executor.cpp

```diff
@@ -270,6 +265,22 @@ Error ScriptRunner::AsyncRunScripts(State state, Action action, HandlerFunction
 	return Execute(scripts_iterator, scripts_iterator_end, handler);
 }
 
+
+Error ScriptRunner::RunScripts(State state, Action action) {
+	auto run_err {error::NoError};
+	auto err {AsyncRunScripts(state, action, [this, &run_err](Error error) {
+		log::Info("Finished running sync script");
+		run_err = error;
+		this->loop_.Stop();
+	})};
+	if (err != error::NoError) {
+		this->loop_.Stop();
```

Comment:
Should not call `Stop` when `Run` hasn't been called yet. I think you can just remove it.